nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CaptureChangeMySQL.java

```diff
@@ -0,0 +1,928 @@
+                            || eventType == EventType.PRE_GA_WRITE_ROWS) {
+
+                        InsertRowsEventInfo eventInfo = new InsertRowsEventInfo(currentTable, timestamp, currentBinlogFile, currentBinlogPosition, event.getData());
+                        currentSequenceId.set(insertRowsWriter.writeEvent(currentSession, eventInfo, currentSequenceId.get()));
+
+                    } else if (eventType == EventType.DELETE_ROWS
+                            || eventType == EventType.EXT_DELETE_ROWS
+                            || eventType == EventType.PRE_GA_DELETE_ROWS) {
+
+                        DeleteRowsEventInfo eventInfo = new DeleteRowsEventInfo(currentTable, timestamp, currentBinlogFile, currentBinlogPosition, event.getData());
+                        currentSequenceId.set(deleteRowsWriter.writeEvent(currentSession, eventInfo, currentSequenceId.get()));
+
+                    } else {
+                        // Update event
+                        UpdateRowsEventInfo eventInfo = new UpdateRowsEventInfo(currentTable, timestamp, currentBinlogFile, currentBinlogPosition, event.getData());
+                        currentSequenceId.set(updateRowsWriter.writeEvent(currentSession, eventInfo, currentSequenceId.get()));
+                    }
+                    break;
+
+                case ROTATE:
+                    // Update current binlog filename
+                    RotateEventData rotateEventData = event.getData();
+                    currentBinlogFile = rotateEventData.getBinlogFilename();
+                    currentBinlogPosition = rotateEventData.getBinlogPosition();
+                    break;
+                default:
+                    break;
+            }
+
+            // Advance the current binlog position. This way if no more events are received and the processor is stopped, it will resume after the event that was just processed.
+            // We always get ROTATE and FORMAT_DESCRIPTION messages no matter where we start (even from the end), and they won't have the correct "next position" value, so only
+            // advance the position if it is not that type of event.
+            if (eventType != EventType.ROTATE && eventType != EventType.FORMAT_DESCRIPTION) {
+                currentBinlogPosition = header.getNextPosition();
+            }
+
+            // Update state with latest values
+            Map<String, String> newStateMap = new HashMap<>(stateManager.getState(Scope.LOCAL).toMap());
+
+            // Save current binlog filename and position to the state map
+            if (currentBinlogFile != null) {
+                newStateMap.put(BinlogEventInfo.BINLOG_FILENAME_KEY, currentBinlogFile);
+            }
+            newStateMap.put(BinlogEventInfo.BINLOG_POSITION_KEY, Long.toString(currentBinlogPosition));
+            newStateMap.put(EventWriter.SEQUENCE_ID_KEY, String.valueOf(currentSequenceId.get()));
+            stateManager.setState(newStateMap, Scope.LOCAL);
```

Comment:
> binlog position and file are in each flow file

I agree with that it can be recoverable by looking at the latest FlowFile. Then I prefer updating State only when the processor stops. However it may depend on use case, both way have benefit. Can we make it configurable by adding one more processor property?